libraries/bsg_manycore_loader.cpp

```diff
@@ -179,13 +179,30 @@ static int hb_mc_loader_eva_write(const Elf32_Phdr *phdr,
                                   const hb_mc_eva_map_t *map,
                                   hb_mc_coordinate_t tile)
 {
-        hb_mc_eva_t eva = start_eva;
-        size_t off = 0, rem = sz;
+        hb_mc_npa_t start_npa;
+        size_t npa_sz = 0;
         int rc;
         char segname[64];
 
         hb_mc_loader_segment_to_string(phdr, segname, sizeof(segname));
 
+        rc = hb_mc_eva_to_npa(mc, map, &tile, &start_eva, &start_npa, &npa_sz);
+        if(rc != HB_MC_SUCCESS){
+                bsg_pr_err("%s: Failed to translate EVA into a NPA\n",
+                           __func__);
+                return rc;
+        }
+
+        if (hb_mc_manycore_npa_is_dram(mc, &start_npa) &&
+            hb_mc_manycore_dram_is_enabled(mc)) {
+                rc = hb_mc_manycore_eva_write_dma(mc, map, &tile,
+                                                  &start_eva, data, sz);
+                if (rc != HB_MC_NOIMPL) {
```

Comment:
I think this needs to be an if-else.

Reply:
We return either way, either success or error, expect for NOIMPL, in which case, we fall back. So, else is not needed...?

Reply:
But what if you succeed?

Reply:
This code says if you get success you will also fall through and call eva_write.

Reply:
Oh I see. NM. Confusing code style though - you should add a comment if nothing else.